.pylintrc

```diff
@@ -1,265 +1,313 @@
-    no-else-return,     # relax "elif" after a clause with a return
-    missing-yield-doc,      # in coroutines, these checks can yield false
-    missing-yield-type-doc,  # positives (pun intended)
-    import-outside-toplevel,
-    docstring-first-line-empty,
-    no-name-in-module,  # remove when pylint behaves
-    import-error,    # remove when pylint behaves
-    bad-continuation, bad-whitespace # differences of opinion with black
+# no Warning level messages displayed, use "--disable=all --enable=classes
+# --disable=W".
+disable=arguments-renamed,              # TODO: investigate / re-enable
+        bad-mcs-classmethod-argument,   # TODO: investigate / re-enable
+        consider-iterating-dictionary,  # TODO: investigate / re-enable
+        consider-using-dict-items,      # TODO: investigate / re-enable
+        consider-using-f-string,        # TODO: investigate / re-enable
+        consider-using-with,            # TODO: investigate / re-enable
+        differing-param-doc,            # TODO: investigate / re-enable
+        docstring-first-line-empty,     # TODO: investigate / re-enable
+        duplicate-code,                 # too verbose
+        f-string-without-interpolation, # TODO: investigate / re-enable
+        fixme,                          # avoid that to do annotations show up as warnings
+        invalid-name,                   # TODO: investigate / re-enable
+        missing-param-doc,              # TODO: investigate / re-enable (issues with kwargs docs)
+        missing-type-doc,               # TODO: investigate / re-enable
+        no-else-return,                 # relax "elif" after a clause with a return
+        no-member,                      # TODO: investigate / re-enable
+        no-self-use,                    # too verbose
+        protected-access,               # we don't strictly follow the public vs. private convention
+        raise-missing-from,             # TODO: investigate / re-enable
+        redundant-returns-doc,          # TODO: investigate / re-enable
+        super-with-arguments,           # TODO: investigate / re-enable
+        too-few-public-methods,         # too verbose
+        too-many-ancestors,             # too verbose
+        too-many-arguments,             # too verbose
+        too-many-branches,              # too verbose
+        too-many-instance-attributes,   # too verbose
+        too-many-lines,                 # too verbose
+        too-many-locals,                # too verbose
+        too-many-nested-blocks,         # too verbose
+        too-many-public-methods,        # too verbose
+        too-many-statements,            # too verbose
+        unnecessary-pass,               # allow for methods with just "pass", for clarity
+        unspecified-encoding,           # TODO: investigate / re-enable
+        use-dict-literal,               # TODO: investigate / re-enable
+        use-list-literal,               # TODO: investigate / re-enable
```

Comment:
@rathishcholarajan - Do you agree if I would open a follow-up issue to investigate and/or re-enable some of the rules again?

My feeling is that using an up-to-date linter will help developers that are not Python professionals yet (like myself) to stick with best practices.